src/transformers/trainer_seq2seq.py

```diff
@@ -67,14 +110,7 @@ def evaluate(
             dictionary also contains the epoch number which comes from the training state.
         """
 
-        gen_kwargs = gen_kwargs.copy()
-        if gen_kwargs.get("max_length") is None and gen_kwargs.get("max_new_tokens") is None:
```

Comment:
The lines below, setting `self.args.generation_max_length` and `self.args.generation_num_beams` need to be added back, otherwise we will be breaking retrocompatibility :)